### PR TITLE
Support multiple destinations

### PR DESCRIPTION
**Pull request by hiroebe**
I added a feature of multiple destinations, which enables users to write a JSON like this:

```json
{
  "src1": "/dst1",
  "src2": ["/dst2-1", "/dst2-2"]
}
```

For example, it's useful when putting `vimrc` for both vim and neovim:

```json
{
  "vimrc": ["~/.vimrc", "~/.config/nvim/init.vim"]
}
```

### CODE REVIEW

**Comment by rhysd** [comment]
Makes sense. Thanks for adding this. I'm reviewing.

File: src/mappings.go [hunk 1]
```diff
@@ -121,18 +128,33 @@ func convertMappingsJSONToMappings(json MappingsJSON) (Mappings, error) {
...
+				// Note: Ignore if dist is specified 'null' in JSON
+				continue
+			}
+			p, err := expandPath(v)
+			if err != nil {
+				return nil, err
+			}
+			m[k] = []abspath.AbsPath{p}
+		case []interface{}:
+			m[k] = make([]abspath.AbsPath, 0, len(v))
```
Comment (rhysd) [inline src/mappings.go:143]: Please make a temporary variable and then finally substitute it to `m[k]`. Current implementation causes hash look up multiple times.

```go
a := make([]abspath.AbsPath, 0, len(v))

for _, iface := range v {
    // ...
}

m[k] = a
```

File: src/mappings.go [hunk 2]
```diff
@@ -121,18 +128,33 @@ func convertMappingsJSONToMappings(json MappingsJSON) (Mappings, error) {
...
+			for _, iface := range v {
+				s, ok := iface.(string)
+				if !ok {
+					return nil, fmt.Errorf("Type of value must be string or string[]: %v", v)
+				}
```
Comment (rhysd) [inline src/mappings.go:148]: `expandPath` will crash when the value is empty string. Please check `s` is empty or net before calling it like above `case string:` clause.

File: src/mappings_test.go [hunk 3]
```diff
@@ -418,8 +432,11 @@ func TestLinkSourceNotExist(t *testing.T) {
 	}
 }
 
-func TestLinkNullDist(t *testing.T) {
-	m := Mappings{"License.txt": abspath.AbsPath{}}
+func TestLinkNullDest(t *testing.T) {
+	m := Mappings{
+		"empty":     []abspath.AbsPath{},
+		"null_only": []abspath.AbsPath{abspath.AbsPath{}},
+	}
```
Comment (rhysd) [inline src/mappings_test.go:439]: Please add test case for converting `MappingsJSON` to `Mappings` with these edge cases. It will catch mistakes such as https://github.com/rhysd/dotfiles/pull/7#discussion_r467727679.

File: src/mappings.go [hunk 4]
```diff
@@ -27,8 +27,8 @@ func (err NothingLinkedError) Error() string {
 // UnixLikePlatformName is a special platform name used commonly for Unix-like platform
 const UnixLikePlatformName = "unixlike"
 
-type Mappings map[string]abspath.AbsPath
-type MappingsJSON map[string]string
+type Mappings map[string][]abspath.AbsPath
+type MappingsJSON map[string]interface{}
```
[the remark this answers is not in the document]
Comment (hiroebe) [reply src/mappings.go]: I changed `interface{}` to `[]string` for `MappingsJSON`.
However, I think I have to use `interface{}` when decoding json, and it's used [here](https://github.com/rhysd/dotfiles/pull/7/files#diff-4954e3d3c579a6d641c7c1f1c344eb86L98-R110).

In my opinion, moving `interface{}` type into one function is enough.
If you have a better solution for this, let me know.

File: src/mappings.go (hunk 1 above)
Comment (hiroebe) [reply src/mappings.go]: I moved these operations to [here](https://github.com/rhysd/dotfiles/pull/7/files#diff-4954e3d3c579a6d641c7c1f1c344eb86R112-R128), fixing this point.

File: src/mappings.go (hunk 2 above)
Comment (hiroebe) [reply src/mappings.go]: Fixed.
https://github.com/rhysd/dotfiles/pull/7/files#diff-4954e3d3c579a6d641c7c1f1c344eb86R144-R146

File: src/mappings_test.go (hunk 3 above)
Comment (hiroebe) [reply src/mappings_test.go:439]: I added this case.
https://github.com/rhysd/dotfiles/pull/7/files#diff-efaf99ab42d0924fac6876b60948027cR589-R606

**Comment by hiroebe** [comment]
@rhysd Thank you for reviewing this!
I fixed all you pointed out. Please review them again 🙏

File: src/mappings.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (rhysd) [reply src/mappings.go]: ok, sounds good. I think it's reasonable to compromise here.

**Comment by rhysd** [comment]
New release v0.2.1 includes this feature.

**Comment by hiroebe** [comment]
Thank you!
